src/commands/moveToNextHeadingLevelN.ts

```diff
@@ -0,0 +1,24 @@
+import { HeadingLevel, VirtualCommandArgs } from "./types";
+import { AccessibilityNode } from "../createAccessibilityTree";
+
+export type MoveToNextHeadingLevelNArgs = VirtualCommandArgs;
+
+export function moveToNextHeadingLevelN({ headingLevel }: { headingLevel: HeadingLevel }) {
+  const headingElementName = `H${headingLevel}`;
+  const isHeadingLevelN =
+    (node: AccessibilityNode) => node.node.nodeName === headingElementName;
```

Comment:
Care needed here because it is possible (though ill-advised typically) to use an explicit role to create a heading, e.g.

```
<div role="heading" aria-level="1">Heading Level 1<div>
```

and we’ll need to support this as well.

We have the likes of https://github.com/guidepup/virtual-screen-reader/blob/main/src/commands/getNextIndexByRole.ts already for getting the next member of the tree that matches the role, we could probably extend to also allow choosing based on attributes (in this case the level) to give us what we need here in such a way that supports native heading elements as well as aria based ones.

Reply:
Good shout. In my dreams I'm imagining `getNextIndexByRole` could be replaced with something like:

```typescript
function getNextIndex(filters: {
  roles?: Role[],
  attributes?: { [key: string]: AttributeMatcher
}) {
  // ...
}

// If `string`, attribute matches against exact value. Otherwise, attribute matches against predicate.
type AttributeMatcher = string | (value: string) => boolean;
```

This looks like a solid solution to me. What do you think? Does that make sense? Would it still be a good idea to hold on to the `getNextIndexByRole` API anyway for backwards compatibility, or are you happy with the API changing?

Reply:
Examples of how that might be used:

```typescript
const nextBannerRole = getNextIndex({ roles: ['banner']] });
const nextLevel3Heading = getNextIndex({ roles: ['heading'], attributes: { 'heading-level': '3' });
const nextNonTopLevelHeading = getNextIndex({
  roles: ['heading'],
  attributes: { 'heading-level': (headingLevel) => Number(headingLevel) > 1 },
});
```

Reply:
This is an internal method so no harm / risk to updating the interface provided we correctly update everywhere it's used. What you suggest feels reasonable.